server/src/catalog.rs

```diff
@@ -90,8 +98,19 @@ impl Catalog {
             }
         }
 
+        // Add model tables using the data from the model_table_metadata database table.
+        let mut new_model_table_metadata = vec![];
```

Comment:
It was mainly inspired by the initialization of `table_metadata` and `model_table_metadata` which is done in the exact same way in the same function. However, I do agree that it would be better to use the other structure so I have changed it.

Reply:
If I recall correctly that was done so `try_adding_table_or_model_table_metadata()` could access both vectors. However, since `get_new_model_table_metadata()` returns all of the new model tables in one call it may as well allocate the vector.